gobblin-metrics/src/main/java/gobblin/metrics/KafkaReporter.java

```diff
@@ -0,0 +1,426 @@
+    metrics.add(serializeValue(name, meter.getOneMinuteRate(), "rate", "1m"));
+    metrics.add(serializeValue(name, meter.getFiveMinuteRate(), "rate", "5m"));
+    metrics.add(serializeValue(name, meter.getFifteenMinuteRate(), "rate", "15m"));
+
+    return metrics;
+  }
+
+  /**
+   * Serialize a {@link com.codahale.metrics.Snapshot}
+   * @param name
+   * @param snapshot
+   * @return List of byte arrays for each metric derived from the snapshot object
+   */
+  protected List<byte[]> serializeSnapshot(String name, Snapshot snapshot) {
+    List<byte[]> metrics = new ArrayList<byte[]>();
+
+    metrics.add(serializeValue(name, snapshot.getMean(), "mean"));
+    metrics.add(serializeValue(name, snapshot.getMin(), "min"));
+    metrics.add(serializeValue(name, snapshot.getMax(), "max"));
+    metrics.add(serializeValue(name, snapshot.getMedian(), "median"));
+    metrics.add(serializeValue(name, snapshot.get75thPercentile(), "75percentile"));
+    metrics.add(serializeValue(name, snapshot.get95thPercentile(), "95percentile"));
+    metrics.add(serializeValue(name, snapshot.get99thPercentile(), "99percentile"));
+    metrics.add(serializeValue(name, snapshot.get999thPercentile(), "999percentile"));
+
+    return metrics;
+  }
+
+  /**
+   * Serialize a single value
+   * @param name
+   * @param value
+   * @param path suffixes to more precisely identify the meaning of the reported value
+   * @return Singleton list of byte arrays representing the value
+   */
+  protected List<byte[]> serializeSingleValue(String name, Object value, String... path) {
+    List<byte[]> metrics = new ArrayList<byte[]>();
+    metrics.add(serializeValue(name, value, path));
+    return metrics;
+  }
+
+  /**
+   * Serializes a single metric key-value pair to send to Kafka
+   * @param name name of the metric
+   * @param value value of the metric to report
+   * @param path additional suffixes to further identify the meaning of the reported value
+   * @return a byte array containing the key-value pair representing the metric
+   */
+  protected byte[] serializeValue(String name, Object value, String... path) {
+    String str = stringifyValue(name, value, path);
+    if(str != null) {
+      return str.getBytes();
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Converts a single metric key-value pair to a string
+   * @param name name of the metric
+   * @param value value of the metric to report
+   * @param path additional suffixes to further identify the meaning of the reported value
+   * @return a string containing the key-value pair representing the metric
+   */
+  protected synchronized  String stringifyValue(String name, Object value, String... path) {
+    Metric metric = new Metric();
+    metric.name = makeName(name, path);
+    metric.value = value;
+    metric.tags = tags;
+    metric.timestamp = System.currentTimeMillis();
+
+    try {
+      return mapper.writeValueAsString(metric);
```

Comment:
OK, this is probably why you want to have the `Metric` class. In which format does this mapper write?